django_camunda/dmn/utils.py

```diff
@@ -63,3 +65,57 @@ def parse_dmn(xml: str) -> IntrospectionResult:
     inputs = [process_input_var(input_var) for input_var in input_vars]
     outputs = dict((process_output_var(output_var) for output_var in output_vars))
     return IntrospectionResult(inputs=inputs, output=outputs)
+
+
+def extract_decision_definition_inputs(
+    xml: _Element, definition_id: str
+) -> List[DMNInputParameter]:
+    decision_table = xml.find(
+        f".//decision[@id='{definition_id}']", namespaces=xml.nsmap
+    )
+    input_elements = decision_table.findall(".//input", namespaces=xml.nsmap)
+    dependency_tables = [
+        dependency.attrib["href"].lstrip("#")
+        for dependency in decision_table.findall(
+            ".//requiredDecision", namespaces=xml.nsmap
+        )
+    ]
+
+    input_vars = []
+    for input_element in input_elements:
+        expression_element = input_element.find("inputExpression", namespaces=xml.nsmap)
+        text_element = expression_element.find("text", namespaces=xml.nsmap)
+        input_expression = text_element.text
+        if input_expression in dependency_tables:
+            input_vars += extract_decision_definition_inputs(xml, input_expression)
+            continue
+
+        input_vars.append(
+            DMNInputParameter(
+                id=input_element.attrib["id"],
+                label=input_element.attrib["label"],
+                expression=input_expression,
+                type_ref=expression_element.attrib["typeRef"],
+            )
+        )
+    return input_vars
+
+
+def extract_decision_definition_outputs(
+    xml: _Element, definition_id: str
+) -> List[DMNOutputParameter]:
+    decision_table = xml.find(
+        f".//decision[@id='{definition_id}']", namespaces=xml.nsmap
+    )
+    output_elements = decision_table.findall(".//output", namespaces=xml.nsmap)
+
+    output_vars = [
+        DMNOutputParameter(
+            id=element.attrib["id"],
+            label=element.attrib["label"],
+            name=element.attrib["name"],
+            type_ref=element.attrib["typeRef"],
+        )
+        for element in output_elements
+    ]
+    return output_vars
```

Comment:
```suggestion
    return [
        DMNOutputParameter(
            id=element.attrib["id"],
            label=element.attrib["label"],
            name=element.attrib["name"],
            type_ref=element.attrib["typeRef"],
        )
        for element in output_elements
    ]
```
Really small performance improvement I guess